# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/telemetry/Histograms.json
# Commit: 373bc48d5325
# Full Hash: 373bc48d5325e17cb67a11035d4542d221023594
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2023-09-19 09:37:28
# Regressor Bug: 1784261
# File Overlap Count: 1
# Description:
#   Bug 1784261 - Add telemetry to understand the reason of connection closure, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D188132
# ==============================================================================

diff -r f2fee3884997 -r 373bc48d5325 toolkit/components/telemetry/Histograms.json
--- a/toolkit/components/telemetry/Histograms.json	Mon Sep 18 14:28:30 2023 +0000
+++ b/toolkit/components/telemetry/Histograms.json	Mon Sep 18 14:44:18 2023 +0000
@@ -17723,6 +17723,18 @@
     "kind": "enumerated",
     "description": "The reason why a HTTP transaction is restarted."
   },
+  "HTTP_CONNECTION_CLOSE_REASON": {
+    "record_in_processes": ["main", "socket"],
+    "products": ["firefox"],
+    "alert_emails": ["necko@mozilla.com", "kershaw@mozilla.com"],
+    "expires_in_version": "never",
+    "kind": "enumerated",
+    "keyed": true,
+    "n_values": 50,
+    "bug_numbers": [1784261],
+    "releaseChannelCollection": "opt-out",
+    "description": "Reason for HTTP connection closure, keyed by properties like HTTP version and connection state."
+  },
   "TRANSACTION_ECH_RETRY_WITH_ECH_COUNT": {
     "record_in_processes": ["main", "socket"],
     "products": ["firefox"],
